### PR TITLE
Change MeshBase::spatial_dimension()

### PR DESCRIPTION
**Pull request by jwpeterson**
Initial attempt as per the discussion in #756.  I suspect this will need a couple iterations before it is exactly right, but this basically does what I think it is supposed to do.  Would be great if @roystgnr could comment, especially regarding the tests for x-aligned and planar x-y meshes.  I'll close #756 and put a pointer to this PR there.

### CODE REVIEW

File: src/mesh/nemesis_io_helper.C [hunk 1]
```diff
@@ -917,7 +917,7 @@ void Nemesis_IO_Helper::write_exodus_initialization_info(const MeshBase& pmesh,
 {
   // This follows the convention of Exodus: we always write out the mesh as LIBMESH_DIM-dimensional,
   // even if it is 2D...
-  this->num_dim = pmesh.spatial_dimension();
+  this->num_dim = LIBMESH_DIM;
```
Comment (roystgnr) [inline src/mesh/nemesis_io_helper.C]: Is this what we want in the long run, or just an attempt to keep exodiff tests from failing?

**Comment by roystgnr** [comment]
I'd have said :+1: if MooseBuild hadn't disagreed...

File: src/mesh/nemesis_io_helper.C (hunk 1 above)
Comment (jwpeterson) [inline src/mesh/nemesis_io_helper.C]: Definitely a short-term fix to maintain the original behavior.  Longer term, the right thing to do is let the user choose which dimension to use when writing the mesh, similarly to ExodusII_IO.

**Comment by jwpeterson** [comment]
> I'd have said :+1: if MooseBuild hadn't disagreed...

Yeah, I thought this might happen.  The exodiffs are all of the "Dimension doesn't agree" variety, and I think they are all on 1D problems that are now being written with `num_dim=1` where before they were being written with `LIBMESH_DIM`.  I should be able to fix this with a MOOSE patch that doesn't require a libmesh update...

**Comment by jwpeterson** [comment]
Closing this for now as I don't think I'll be able to merge it before first fixing something in MOOSE.  Will re-open when that is more squared away.

**Pull request closed without merging**